driven/storage/adapter_v3.go

```diff
@@ -11,13 +11,154 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+// FindGroupsV3 finds groups with filter
+func (sa *Adapter) FindGroupsV3(clientID string, filter *model.GroupsFilter) ([]model.Group, error) {
```

Comment:
I don't think that filter should be a pointer since I do not believe we would want to allow this function to be called with a nil filter. If we do want to support this, we would need to handle the nil case first so we don't dereference a nil pointer below.

Reply:
It may be invoked in any kind of configuration with and without a filter. Prefer to check for nil. Thanks for the catch